src/ai/defaultai.cc

```diff
@@ -6754,9 +6851,208 @@ void DefaultAI::update_player_stat(const Time& gametime) {
 					}
 				}
 
-				player_statistics.add(pn, j, me->team_number(), this_player->team_number(),
-				                      cur_strength, old_strength, old60_strength, cass, cur_land,
-				                      old_land, old60_land);
+				// determine the diplomacy score of each player
+				int32_t diplo_score = 0;
+				// we need to be sure all magic numbers have been initialized so wait 30s
+				if (gametime > Time(30000)) {
```

Comment:
I think the diploscore calculation should be skipped for defeated players (if either the calculating or the calculated player is defeated) and probably for the current player too, because I don't think the score for self is used for anything, nor that it would be needed for team calculations. And it doesn't look reliable either.

Reply:
ok good idea

Reply:
done